Bugzilla/Install/Requirements.pm

```diff
@@ -81,471 +72,140 @@ use constant APACHE => qw(apachectl httpd apache2 apache);
-        module  => 'Auth::GoogleAuth',
-        version => '1.01',
-        feature => ['mfa'],
-    },
-    {
-        package => 'GD-Barcode-QRcode',
-        module  => 'GD::Barcode::QRcode',
-        version => '0',
-        feature => ['mfa'],
-    },
-    # Documentation
-    {
-        package => 'File-Copy-Recursive',
-        module  => 'File::Copy::Recursive',
-        version => 0,
-        feature => ['documentation'],
-    }
-    );
-
-    my $extra_modules = _get_extension_requirements('OPTIONAL_MODULES');
-    push(@modules, @$extra_modules);
-    return \@modules;
-};
-
 # This maps features to the files that require that feature in order
 # to compile. It is used by t/001compile.t and mod_perl.pl.
 use constant FEATURE_FILES => (
     jsonrpc       => ['Bugzilla/WebService/Server/JSONRPC.pm', 'jsonrpc.cgi'],
     xmlrpc        => ['Bugzilla/WebService/Server/XMLRPC.pm', 'xmlrpc.cgi',
                       'Bugzilla/WebService.pm', 'Bugzilla/WebService/*.pm'],
-    rest          => ['Bugzilla/WebService/Server/REST.pm', 'rest.cgi'],
+    rest          => ['Bugzilla/API/Server.pm', 'rest.cgi', 'Bugzilla/API/*/*.pm',
+                      'Bugzilla/API/*/Server.pm', 'Bugzilla/API/*/Resource/*.pm'],
+    psgi          => ['app.psgi'],
     moving        => ['importxml.pl'],
     auth_ldap     => ['Bugzilla/Auth/Verify/LDAP.pm'],
     auth_radius   => ['Bugzilla/Auth/Verify/RADIUS.pm'],
+    documentation => ['docs/makedocs.pl'],
     inbound_email => ['email_in.pl'],
     jobqueue      => ['Bugzilla/Job/*', 'Bugzilla/JobQueue.pm',
                       'Bugzilla/JobQueue/*', 'jobqueue.pl'],
     patch_viewer  => ['Bugzilla/Attachment/PatchReader.pm'],
     updates       => ['Bugzilla/Update.pm'],
-    memcached     => ['Bugzilla/Memcache.pm'],
     mfa           => ['Bugzilla/MFA/*.pm'],
+    markdown      => ['Bugzilla/Markdown.pm'],
+    memcached     => ['Bugzilla/Memcache.pm'],
+    auth_delegation => ['auth.cgi'],
 );
 
-# This implements the REQUIRED_MODULES and OPTIONAL_MODULES stuff
-# described in in Bugzilla::Extension.
-sub _get_extension_requirements {
-    my ($function) = @_;
-
-    my $packages = extension_requirement_packages();
-    my @modules;
-    foreach my $package (@$packages) {
-        if ($package->can($function)) {
-            my $extra_modules = $package->$function;
-            push(@modules, @$extra_modules);
-        }
+sub load_cpan_meta {
+    my $dir = bz_locations()->{libpath};
+    my @meta_json = map { File::Spec->catfile($dir, $_) } qw( MYMETA.json META.json );
+    my ($file) = grep { -f $_ } @meta_json;
+
+    if ($file) {
+        open my $meta_fh, '<', $file or die "unable to open $file: $!";
+        my $str = do { local $/ = undef; scalar <$meta_fh> };
+        # detaint
+        $str =~ /^(.+)$/s; $str = $1;
+        close $meta_fh;
+
+        return CPAN::Meta->load_json_string($str);
     }
-    return \@modules;
-};
+    else {
+        ThrowCodeError('cpan_meta_missing');
```

Comment:
Need to load Bugzilla::Error at top otherwise it dies not able to find ThrowCodeError()